codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/transport/RawWebSocketTransport.java

```diff
@@ -0,0 +1,240 @@
+import io.netty.handler.codec.sockjs.SockJsConfig;
+import io.netty.handler.codec.sockjs.SockJsSessionContext;
+import io.netty.handler.codec.sockjs.SockJsService;
+import io.netty.handler.codec.sockjs.handler.SessionHandler.Event;
+import io.netty.handler.codec.sockjs.handler.SockJsHandler;
+import io.netty.util.AttributeKey;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * WebSocketTransport is responsible for the WebSocket handshake and
+ * also for receiving WebSocket frames.
+ */
+public class RawWebSocketTransport extends ChannelInboundHandlerAdapter {
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(RawWebSocketTransport.class);
+    private static final AttributeKey<HttpRequest> REQUEST_KEY = AttributeKey.valueOf("raw.ws.request.key");
+    private final SockJsConfig config;
+    private final String protocols;
+    private final SockJsService service;
+    private WebSocketServerHandshaker handshaker;
+
+    public RawWebSocketTransport(final SockJsConfig config, final SockJsService service) {
+        checkNotNull(config, "config");
+        checkNotNull(service, "service");
+        this.config = config;
+        protocols = protocolsAsString(config.webSocketProtocols());
+        this.service = service;
+    }
+
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        if (msg instanceof FullHttpRequest) {
+            handleHttpRequest(ctx, (FullHttpRequest) msg);
+        } else if (msg instanceof WebSocketFrame) {
+            handleWebSocketFrame(ctx, (WebSocketFrame) msg);
+        } else {
+            ctx.fireChannelRead(msg);
+        }
+    }
+
+    private static boolean checkRequestHeaders(final ChannelHandlerContext ctx, final HttpRequest req) {
+        if (req.method() != GET) {
+            ctx.writeAndFlush(responseFor(req)
+                    .methodNotAllowed()
+                    .header(CONTENT_LENGTH, 0)
+                    .allow(GET)
+                    .buildResponse())
+            .addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+
+        final String upgradeHeader = req.headers().get(UPGRADE);
+        if (upgradeHeader == null || !"websocket".equals(upgradeHeader.toLowerCase())) {
+            ctx.writeAndFlush(responseFor(req)
+                    .badRequest()
+                    .content("Can \"Upgrade\" only to \"WebSocket\".")
+                    .contentType(HttpResponseBuilder.CONTENT_TYPE_PLAIN)
+                    .buildFullResponse()).addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+
+        String connectHeader = req.headers().get(CONNECTION);
+        if (connectHeader != null && "keep-alive".equals(connectHeader.toLowerCase())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1157)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) When doing a String.toLowerCase()/toUpperCase() call, use a Locale [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseLocaleWithCaseConversions)